pkg/model/resource/resource.go

```diff
@@ -19,54 +19,113 @@ package resource
-	GroupPackageName string `json:"-"`
-
-	// Version is the API version.
+	// Domain is the resource's domain.
+	Domain string `json:"domain,omitempty"`
+	// Version is the resource's version.
 	Version string `json:"version,omitempty"`
-
-	// Kind is the API Kind.
+	// Kind is the resource's kind.
 	Kind string `json:"kind,omitempty"`
 
-	// Plural is the API Kind plural form.
+	// Plural is the resource's kind plural form.
 	Plural string `json:"plural,omitempty"`
 
-	// ImportAlias is a cleaned concatenation of Group and Version.
-	ImportAlias string `json:"-"`
+	// Path is the path to the go package where the types are defined.
+	Path string `json:"path,omitempty"`
 
-	// Package is the go package of the Resource.
-	Package string `json:"package,omitempty"`
+	// API holds the information related to the resource API.
+	API *API `json:"api,omitempty"`
 
-	// Domain is the Group + "." + Domain of the Resource.
-	Domain string `json:"domain,omitempty"`
+	// Controller specifies if a controller has been scaffolded.
+	Controller bool `json:"controller,omitempty"`
 
-	// Namespaced is true if the resource is namespaced.
-	Namespaced bool `json:"namespaced,omitempty"`
+	// Webhooks holds the information related to the associated webhooks.
+	Webhooks *Webhooks `json:"webhooks,omitempty"`
+}
+
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (r Resource) QualifiedGroup() string {
+	switch "" {
+	case r.Domain:
+		return r.Group
+	case r.Group:
+		return r.Domain
+	default:
+		return fmt.Sprintf("%s.%s", r.Group, r.Domain)
+	}
+}
+
+// PackageName returns a name valid to be used por go packages.
+func (r Resource) PackageName() string {
+	if r.Group == "" {
+		return safeImport(r.Domain)
+	}
+
+	return safeImport(r.Group)
+}
 
-	// CRDVersion holds the CustomResourceDefinition API version used for the Resource.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the Resource.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
+// ImportAlias returns a identifier usable as an import alias for this resource.
+func (r Resource) ImportAlias() string {
+	if r.Group == "" {
+		return safeImport(r.Domain + r.Version)
+	}
+
+	return safeImport(r.Group + r.Version)
+}
+
+// GVK returns the GVK identifier of a resource.
+func (r Resource) GVK() GVK {
+	return GVK{
+		Group:   r.QualifiedGroup(),
+		Version: r.Version,
+		Kind:    r.Kind,
+	}
 }
 
-// GVK returns the group-version-kind information to check against tracked resources in the configuration file
-func (r *Resource) GVK() config.GVK {
-	return config.GVK{
-		Group:          r.Group,
-		Version:        r.Version,
-		Kind:           r.Kind,
-		CRDVersion:     r.CRDVersion,
-		WebhookVersion: r.WebhookVersion,
+// Update combines fields of two resources that have matching GVK favoring the receiver's values.
+func (r *Resource) Update(other *Resource) error {
```

Comment:
```suggestion
func (r *Resource) Update(other *Resource) {
```
The error returned here was added only because r is a pointer and it can be nil. Then,  if r is nil we ought to only return. Has no need to create a new error and raise it in this scenario.